README.md

```diff
@@ -17,83 +17,318 @@ Main characteristics:
+
+  - **Dynamic Dependency Listening:** Dependencies are determined based on their usage. This implies that a dependency store not actively used is not automatically "listened" to, optimizing resource utilization.
+
+#### Switch map
+
+This capability to subscribe/unsubscribe to the dependency stores allows to create switch maps in a natural way.
+
+```typescript
+import {writable, computed} from "@amadeus-it-group/tansu";
+
+const switchToA$ = writable(true);
+const a$ = writable(1);
+const b$ = writable(0);
+
+const computedValue$ = computed(() => {
+  if (switchToA$()) {
+    console.log('Return a$');
+    return a$();
+  } else {
+    console.log('Return b$');
+    return b$();
+  }
+});
+
+computedValue$.subscribe((value) => console.log('Computed value:', value));
+a$.set(2);
+switchToA$.set(false);
+a$.set(3);
+a$.set(4);
+switchToA$.set(true);
+
+```
+
+output:
+
+```text
+Return a$
+Computed value: 1
+Return a$
+Computed value: 2
+Return b$
+Computed value: 0
+Return a$
+Computed value: 4
+```
+
+When `switchToA$.set(false)` is called, the subscription to `a$` is canceled, which means that subsequent changes to `a$` will no longer trigger the calculation., which is only performed again when switchToA$ is set back to true.
+
+### readable
+
+[api documentation](https://amadeusitgroup.github.io/tansu/functions/readable.html)
+
+Similar to Svelte stores, this function generates a store where the value cannot be externally modified.
+
+```typescript
+import {readable} from '@amadeus-it-group/tansu';
+
+const time = readable(new Date(), (set) => {
+  const interval = setInterval(() => {
+    set(new Date());
+  }, 1000);
+
+  return () => clearInterval(interval);
+});
+```
+
+### derived vs computed
+
+While derived and computed may appear similar, they exhibit distinct characteristics that can significantly impact effectiveness based on use-cases:
+
+- **Declaration of Dependencies:**
+  - `computed`: No explicit declaration of store dependencies is required, providing more flexibility in code composition.
+  - `derived`: Requires explicit declaration of dependencies.
+
+- **Performance:**
+  - `computed`: Better performance by re-running the function only based on changes in the stores involved in the last run.
+  - `derived`: Re-run the function each time a dependent store changes.
+
+- **Asynchronous State:**
+  - `computed`: Unable to manage asynchronous state.
+  - `derived`: Can handle asynchronous state with the `set` method.
+
+- **Skipping Value Emission:**
+  - `computed`: Does not provide a mechanism to skip emitting values.
+  - `derived`: Allows skipping the emission of values by choosing not to call the provided `set` method.
+
+- **Setup and Teardown:**
+  - `computed`: Lacks explicit setup and teardown methods.
+  - `derived`: Supports setup and teardown methods, allowing actions such as adding or removing DOM listeners.
+
```

Comment:
Note that a specific case related to setup and teardown can be mentioned: if a computed or derived store has a listener that unsubscribes and re-subscribes while the dependencies did not change, the computed is not called again (the previous value is considered up-to-date because the dependencies did not change) whereas the derived is called again (because it may, for example, add or remove DOM listeners).